src/Monitoring/DotNet/Monitoring.EventProcessor/EventHubMonitor.cs

```diff
@@ -0,0 +1,124 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Reactive.Concurrency;
+using System.Reactive.Linq;
+using System.Reactive.Subjects;
+using System.Reactive.Threading.Tasks;
+using System.Threading.Tasks;
+using Microsoft.ServiceBus.Messaging;
+
+namespace Microsoft.Practices.IoTJourney.Monitoring.EventProcessor
+{
+    public class EventHubMonitor : IObservable<PartitionSnapshot>
```

Comment:
It is not normal practice to Implement `IObservable<T>` (is a relationship), instead you would provide access to an `IObservable<T>` instance (has a relationship). The standard way to do this is with `Observable.Create` [http://introtorx.com/Content/v1.0.10621.0/18_UsageGuidelines.html]